lib/smart_proxy_host_reports/ansible_processor.rb

```diff
@@ -82,46 +75,61 @@ def spool_report
 
   private
 
-  def log_level(log)
-    if log["failed"]
-      "err"
-    elsif log["changed"]
-      "notice"
-    else
-      "info"
+  def process_facts(result)
+    # TODO: add fact processing and sending to the fact endpoint
+    result["result"]["ansible_facts"] = {}
+  end
+
+  def process_keywords(result)
+    if result["failed"]
+      add_keywords("HasFailure", "AnsibleTaskFailed:#{result["task"]["action"]}")
+    elsif result["result"]["changed"]
+      add_keywords("HasChange")
     end
   end
 
-  def log_source(log)
-    # Temporary until  is resolved
-    log["module"]
+  def process_level(result)
+    if result["failed"]
+      result["level"] = "err"
+    elsif result["result"]["changed"]
+      result["level"] = "notice"
+    else
+      result["level"] = "info"
+    end
   end
 
-  def log_message(log)
-    return log["msg"] if log["failed"]
+  def process_message(result)
+    msg = "N/A"
+    return msg if result["task"].nil? || result["task"]["action"].nil?
+    return result["result"]["msg"] if result["failed"]
```

Comment:
We don't store the returned value anywhere. Maybe it's better to do like this?
```suggestion
    return result["friendly_message"] = msg if result["task"].nil? || result["task"]["action"].nil?
    return result["friendly_message"] = result["result"]["msg"] if result["failed"]
```